### PR TITLE
feat: shipping method

### PR DESCRIPTION
**Pull request by oleksandr-honcharuk**
shipping method

## Description

implement select shipping method in checkout 

## Related Issue

[AB#61984](https://dev.azure.com/orckestra001/da965986-c85b-43be-9915-219568b000e5/_workitems/edit/61984)
[AB#60291](https://dev.azure.com/orckestra001/da965986-c85b-43be-9915-219568b000e5/_workitems/edit/60291)

## Motivation and Context

second step of checkout, selecting shipping method

## How Has This Been Tested?

select shipping fulfilment method
fill address
click select shipping method

## Screenshots (if appropriate):
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/composables/src/getters/cartGetters.ts [hunk 1]
```diff
@@ -250,6 +251,14 @@ function isReadyForOrder(cart: Cart): boolean {
   return true;
 }
 
+function getShipment(cart: Cart): Shipment {
+  return cart?.shipments?.[0];
+}
```
Comment (InnaBoitsun) [inline packages/composables/src/getters/cartGetters.ts:256]: we already have getActiveShipment(cart)

File: packages/theme/components/Checkout/AddressForm.vue [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+  </div>
+</template>
+
+<script>
+import { SfButton, SfInput, SfSelect, SfCheckbox } from '@storefront-ui/vue';
+import { ValidationObserver, ValidationProvider } from 'vee-validate';
+import { countriesGetters, useCountries } from '@vue-storefront/orc-vsf';
+
+export default {
+  name: 'AddressForm',
```
Comment (InnaBoitsun) [inline packages/theme/components/Checkout/AddressForm.vue:178]: Is this possible to re-use existent ShippingAddressForm.vue ( just rename it to UserAddressForm.vue) ?

File: packages/theme/pages/Checkout/Shipping.vue [hunk 3]
```diff
@@ -232,48 +202,155 @@ export default {
...
+          persist: false,
+          title: 'Checkout process'
+        });
+      } else {
+        onComplete();
+      }
+    };
+
+    const updateShippingMethod = () => {
+    };
```
Comment (InnaBoitsun) [inline packages/theme/pages/Checkout/Shipping.vue:306]: looks like you do not update shipping method in cart when it is selected . need to update it

**Review by InnaBoitsun — CHANGES_REQUESTED** [review]

File: packages/theme/pages/Checkout/Shipping.vue [hunk 4]
```diff
@@ -3,168 +3,147 @@
...
+                      @click="(isOpen = { ...isOpen, [item.id]: !isOpen[item.id] })"
+                    >{{ isOpen[item.id] ? "- info" : "+ info" }}
+                    </SfButton>
+                  </div>
+                </div>
+              </template>
+              <template  #description="{ description }">
+                <div class="sf-radio__description shipping__description">
+                  <transition name="sf-fade">
+                    <div v-if="isOpen[item.id]" class="shipping__info">
```
Comment (InnaBoitsun) [inline packages/theme/pages/Checkout/Shipping.vue:86]: use existent AddressPreview component for detailed info

File: packages/theme/pages/Checkout/Shipping.vue [hunk 5]
```diff
@@ -175,19 +154,14 @@
...
           </SfButton>
           <SfButton
-            v-if="!isFormSubmitted"
-            :disabled="loading"
+            :disabled="loadingFulfillmentMethods || loadingAddresses || loadingCart || (isShipping && !form.addressId)"
             class="form__action-button"
             type="submit"
           >
             {{ $t('Select shipping method') }}
           </SfButton>
```
Comment (InnaBoitsun) [inline packages/theme/pages/Checkout/Shipping.vue:162]: this button should Go to Payment and activated when Shipping is ready

File: packages/theme/components/Checkout/AddressForm.vue (hunk 2 above)
Comment (oleksandr-honcharuk) [reply packages/theme/components/Checkout/AddressForm.vue:177]: ShippingAddressForm has actions inside 
we can implement AddressForm to ShippingAddressForm in feature

File: packages/theme/pages/Checkout/Shipping.vue [hunk 6]
```diff
@@ -229,51 +157,161 @@ export default {
...
+      if (isShipping.value) {
+        updatedShipment.address = isAuthenticated.value ? addresses.value.find(x => x.id === form.value.addressId) : addressForm.value;
+      }
+
+      onUpdate(updatedShipment, () => router.push(context.root.localePath({ name: 'payment' })));
+    };
+
+    onSSR(async () => Promise.allSettled([
+      loadUserShipping(),
+      loadCountries(),
```
Comment (InnaBoitsun) [inline packages/theme/pages/Checkout/Shipping.vue:289]: loadCountries already present in default layout, you can remove it here

File: packages/theme/pages/Checkout/Shipping.vue [hunk 7]
```diff
@@ -229,51 +157,161 @@ export default {
...
-    const isFormSubmitted = ref(false);
-    const { load, save, loading } = useShipping();
+    const { send: sendNotification } = useUiNotification();
+    const { cart, update, error, loading: loadingCart } = useCart();
+    const { addresses, load: loadUserShipping, addAddress, loading: loadingAddresses, error: userAddressError } = useUserAddresses();
+    const { load: loadCountries } = useCountries();
+    const { load: loadFulfillmentMethods, fulfillmentMethods, loading: loadingFulfillmentMethods } = useFulfillmentMethods();
+    const { isAuthenticated } = useUser();
+
+    const shipment = cartGetters.getActiveShipment(cart.value);
```
Comment (InnaBoitsun) [inline packages/theme/pages/Checkout/Shipping.vue:176]: use computed(() =>

**Review by InnaBoitsun — CHANGES_REQUESTED** [review]

File: packages/theme/pages/Checkout/Shipping.vue [hunk 8]
```diff
@@ -3,168 +3,105 @@
...
+                  >
+                    {{ $t('Cancel') }}
+                  </SfButton>-->
+                </div>
+              </form>
+            </ValidationObserver>
+          </template>
+          <template v-else>
+            <SfButton
+              class="action-button sf-button form__action-button--add-address"
```
Comment (InnaBoitsun) [inline packages/theme/pages/Checkout/Shipping.vue:89]: please use class sf-button--text to be as lint

**Pull request merged**